### ISSUE TITLE
can I run this on a headless server with no gpu?

### ISSUE THREAD

**Issue by adrianog** [mscrnt/ubuntu-desktop-docker#1]

**Comment by mscrnt** [comment]
No. It wasn't configured to be headless. There are other docker images that are built for this already. Is there something about this image you like that others don't offer? If not, I suggest checking out another project like this one...

https://github.com/gezp/docker-ubuntu-desktop

**Comment by adrianog** [comment]
I want:
Ubuntu 22
xfce
no gpu required
rdp (x2go/vnc nice to have)
systemd as pid 1

**Comment by mscrnt** [comment]
Does this work for you?

https://github.com/mscrnt/ubuntu-desktop-docker/tree/headless

**Comment by adrianog** [comment]
Yes.
What are the implications of running the container as "privileged" though? is it going to modify the host in any way. e.g. user creation?
Is it used for rw cgroup?
`-v /sys/fs/cgroup:/sys/fs/cgroup:rw:`
see https://github.com/moby/moby/issues/42040
and long discussion, linked from there: https://github.com/moby/moby/issues/42275

PS:
similar comment I made on another repo:
https://github.com/infrastlabs/docker-headless/issues/12

**Comment by mscrnt** [comment]
Running in privileged mode is a little heavy handed. Let me see if I can work out another way that isn't so intrusive... It can definitely modify the host.

**Comment by adrianog** [comment]
hey btw, I'm running the headless container without a gpu card, thanks.

I've tried to get snap running in the container but it's always failing after installing an app it cannot find the file to execute it?
snap list just hangs.

Did you get snap to work?

PS: tried to reboot the container and it doesn't work, looking at docker logs I see:
Failed to attach 1 to compat systemd cgroup /docker/7a16c64022374e09a94363996519ca50c4008fca1207e5456aac259c038e0f8e/init.scope: No such file or directory

**Comment by mscrnt** [comment]
I'll have to investigate this a bit more. I might have some time this week to dig back into this project.

**Comment by mscrnt** [comment]
@adrianog Sorry, it's been a long week.

I think I got this working. I've pushed the changes to the headless branch. Go ahead and give it a try and let me know if it works.

**Comment by mscrnt** [comment]
Main can now be run headless or with GPU support. I've also preinstalled snap. I did not include rdp or x2go in this image, but I think ssh and vnc should be sufficient.

**Comment by adrianog** [comment]
could we possible have rdp? it's a superior experience all around.
Thank you by the way, I'll be testing it as soon as rdp is in.

**Comment by adrianog** [comment]
Would it be possible also to have a different tag to support this branch of xrdp? https://archive.is/m5uz3
Seems to boost performance quite a lot.

**Comment by mscrnt** [comment]
I'm having some trouble on my end getting this to work correctly again. I'll be spending a good amount of time this week troubleshooting. I'll let you know when I make progress with this. Thank you for you patience.

**Comment by mscrnt** [comment]
@adrianog — yes, v1.0.0 covers your original asks: XFCE, no GPU required, xrdp, systemd as PID 1, and no `--privileged` (on cgroup v2 hosts the recipe is just `--cgroupns=host` + a writable `/sys/fs/cgroup` bind + tmpfs).

Snap was dropped (broken inside containers) and standard xrdp is used. Full release notes: https://github.com/mscrnt/ubuntu-desktop-docker/releases/tag/v1.0.0

Image will be on GHCR / Docker Hub shortly — I'll get the registry push sorted in a follow-up.

**Comment by mscrnt** [comment]
Image is live on Docker Hub:

\`\`\`
docker pull mscrnt/ubuntu-desktop:1.0.0
# or slim
docker pull mscrnt/ubuntu-desktop:1.0.0-slim
\`\`\`

Both variants include NoMachine NX (port 4000) — requires \`--cap-add=SYS_PTRACE\` at runtime per [NoMachine KB DT07S00242](https://kb.nomachine.com/DT07S00242).
